osu.Game/Overlays/Direct/DownloadButton.cs

```diff
@@ -1,76 +1,146 @@
 ﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
 // Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
 
+using osu.Framework.Allocation;
 using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
 using osu.Framework.Input;
 using osu.Game.Beatmaps;
 using osu.Game.Beatmaps.Drawables;
 using osu.Game.Graphics;
 using osu.Game.Graphics.Containers;
 using OpenTK;
+using OpenTK.Graphics;
 
 namespace osu.Game.Overlays.Direct
 {
     public class DownloadButton : OsuClickableContainer
     {
         private readonly SpriteIcon icon;
+        private readonly SpriteIcon checkmark;
+        private readonly BeatmapSetDownloader downloader;
+        private readonly Box background;
+
+        private OsuColour colours;
 
         public DownloadButton(BeatmapSetInfo set, bool noVideo = false)
         {
-            BeatmapSetDownloader downloader;
             Children = new Drawable[]
             {
                 downloader = new BeatmapSetDownloader(set, noVideo),
+                new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    CornerRadius = 17,
```

Comment:
There's no way this is right. You should be using a `CircularContainer` to ensure it always maintains the correct rounding values.